src/app/models/AnalysisJob.ts

```diff
@@ -92,21 +141,33 @@ export class AnalysisJob extends AbstractModel implements IAnalysisJob {
   public updater?: User;
   @deleter<AnalysisJob>()
   public deleter?: User;
-  @hasOne<AnalysisJob, Script>(SCRIPT, "scriptId")
-  public script?: Script;
-  @hasOne<AnalysisJob, SavedSearch>(SAVED_SEARCH, "savedSearchId")
-  public savedSearch?: SavedSearch;
+  @hasOne<AnalysisJob, Project>(PROJECT, "projectId")
+  public project?: Project;
+  @hasMany<AnalysisJob, Script>(SCRIPT, "scriptIds")
+  public scripts?: Script[];
+  @hasMany<AnalysisJob, AudioEventImport>(AUDIO_EVENT_IMPORT, "audioEventImportIds")
+  public audioEventImports?: AudioEventImport[];
 
   public constructor(analysisJob: IAnalysisJob, injector?: AssociationInjector) {
     super(analysisJob, injector);
   }
 
-  public get viewUrl(): string {
-    return audioAnalysisMenuItem.route.format({
+  public createViewUrl(fakeProjectId?: Id): string {
+    const projectId = this.projectId ?? fakeProjectId;
+    if (!projectId) {
+      return this.adminViewUrl;
+    }
+
+    return audioAnalysisMenuJobItem.route.format({
       analysisJobId: this.id,
+      projectId,
     });
   }
 
+  public get viewUrl(): string {
+    throw new Error("method not implemented");
```

Comment:
why not implemented? you have all the information needed....

Reply:
Kind of. I previously had a `createViewUrl` method that allowed system analysis jobs to be fake scoped to a project

![image](https://github.com/user-attachments/assets/61181dc9-0dda-4713-a4d2-ff3df6d538a1)

The problem is that system analysis jobs do not have a `project_id` field, so I only have enough information to return the `adminViewUrl` for system analysis jobs

Reply:
well in the common case most jobs will have project id

can view url take an optional parameter?

Reply:
The `viewUrl` is a getter, so it can't take in an optional parameter, hence the `createViewUrl` method. `createViewUrl` is a convention of our models (though not backed up by any TS typing).

Reply:
k, i still reckon a default implementation is better than none